### DISCUSSION TITLE
NAUTILUS EXLORER - Duplicate cards being created after logging out and back in

### DISCUSSION THREAD

**Discussion by cmeffley** [nss-evening-cohort-14/evening-client-side#203] [Help Me!] [answered]
Chris Meffley, Honey-Rae Swan, Matthew Gonzales, Hunter Juneau

Yes: 
#166 , #195 - Tried both, but the did not appear to fix the problem.

When we sign in/out multiple times, duplicate cards are created. They have the correct information in Firebase, just multiple cards are created.

No errors

https://stackoverflow.com/questions/10444077/javascript-removeeventlistener-not-working
https://javascript.info/bubbling-and-capturing
https://learnersbucket.com/examples/javascript/different-ways-to-reload-refresh-page-in-javascript/
https://www.freecodecamp.org/news/location-reload-method-how-to-reload-a-page-in-javascript/

- Tried the answers from Issue tickets #166 and #195
- Added `true` to the event listener and remove event listener
- Added `true` to `window.location.reload(true)`
- Added `window.location.reload` to the logout button
- Tried `window.location.href`

DomEvents:
https://github.com/nss-evening-cohort-14/nautilus-explorer-dashboard-nautilus-explorer-dashboard/blob/development/src/javascripts/events/domEvents.js
Signout:
https://github.com/nss-evening-cohort-14/nautilus-explorer-dashboard-nautilus-explorer-dashboard/blob/development/src/javascripts/helpers/auth/signOut.js
Login Button:
https://github.com/nss-evening-cohort-14/nautilus-explorer-dashboard-nautilus-explorer-dashboard/blob/development/src/javascripts/components/buttons/loginButton.js
Logout Button:
https://github.com/nss-evening-cohort-14/nautilus-explorer-dashboard-nautilus-explorer-dashboard/blob/development/src/javascripts/components/buttons/logoutButton.js

**Comment by cmeffley** [comment, marked as answer]
With the way we are passing `user`, our StartApp was running twice, one with user logged in and one with no one logged in. So our Event listeners were running/listening twice. To correct this, we created two domEvents: one for logged in, and one for logged out. The logged out domEvents no longer have event listeners that create data, so it can't duplicate.

**Comment by danicrosby** [comment]
Can I piggyback on this thread? We too have this problem of cards doubling on login/logout. I read and implemented all of these scenarios into our code with no luck. The one difference we have with our code is that it's written a bit differently than we learned in class (switch statements etc). I tried to 'removeEventListener' like stated in a couple of examples, but end up with a dependency error. 

I had success (for a hot second) with e.preventDefault on our landing page, but that too ultimately failed after multiple logins.

I tried making a separate removeDomEvents file, but again with the dependacy errors

@TrinityChristiana would love to hear your thoughts on a work-around. 

DomEvents: 
https://github.com/nss-evening-cohort-14/clown-college-dashboard-clown-college-dashboard/blob/main/src/javascripts/events/domEvents.js

SignIn:
https://github.com/nss-evening-cohort-14/clown-college-dashboard-clown-college-dashboard/blob/main/src/javascripts/helpers/signIn.js

SignOut: 
https://github.com/nss-evening-cohort-14/clown-college-dashboard-clown-college-dashboard/blob/main/src/javascripts/helpers/signOut.js

Logout/In Button located in NavBar: 
https://github.com/nss-evening-cohort-14/clown-college-dashboard-clown-college-dashboard/blob/main/src/javascripts/components/navbar.js
